tools/jira_client/jira_api.py

```diff
@@ -0,0 +1,34 @@
+from jira.exceptions import JIRAError
+import retry
+
+from .consts import CLOSED_TRANSITION_ID
+
+
+class JiraAPI:
+    def __init__(self, jira_client, logger) -> None:
+        self._jira_client = jira_client
+        self._logger = logger
+
+    def link_issue_to_root_issue(self, issue, root_issue):
+        self._logger.info("Linking issue key=%s to root issue key=%s", issue.key, root_issue.key)
+        self._jira_client.create_issue_link("relates to", issue.key, root_issue.key)
+
+    def close_issue(self, issue):
+        self._logger.info("Was requested to close issue %s ignoring for now", issue)
+        self._jira_client.transition_issue(issue.key, CLOSED_TRANSITION_ID)
+
+    def link_and_close_issue(self, issue, linked_issue=None):
+        if linked_issue:
+            self.link_issue_to_root_issue(issue, linked_issue)
+        self.close_issue(issue)
+
+    @retry.retry(exceptions=JIRAError, tries=3, delay=2)  # being resilient to 401 statuses
+    def get_issue_comments(self, issue):
+        if issue is None:
```

Comment:
why should it be none?

Reply:
We kept the existing logic exactly as it was to avoid issues, we can change it if desired?